compiler/sigmatch.nim

```diff
@@ -2231,8 +2231,15 @@ proc matchesAux(c: PContext, n, nOrig: PNode,
       else:
         m.state = csNoMatch
         return
+    
     if formal.typ.kind == tyVar:
-      if not n.isLValue:
```

Comment:
I would have put this logic into a variant of ``isLValue``, but I'm merging it now since it's blocking you.